bolt-modules/boltlib/lib/puppet/datatypes/result.rb

```diff
@@ -1,20 +1,40 @@
 # frozen_string_literal: true
 
 Puppet::DataTypes.create_type('Result') do
-  interface <<-PUPPET
-    attributes => {
-      'value' => Hash[String[1], Data],
-      'target' => Target
-    },
-    functions => {
-      error => Callable[[], Optional[Error]],
-      message => Callable[[], Optional[String]],
-      action => Callable[[], String],
-      to_data => Callable[[], Hash],
-      ok => Callable[[], Boolean],
-      '[]' => Callable[[String[1]], Data]
-    }
-  PUPPET
+  # rubocop:disable Style/GlobalVars
+  if $future
```

Comment:
Is there any breaking change here?

Reply:
Oh, I guess not. I think I got too excited about `$future`. Will update.